### PR TITLE
[processor/transform] introduce aggregate_on_attribute_value function for metrics

### PR DESCRIPTION
**Pull request by odubajDT**
**Link to tracking Issue:** #16224 

**Changes:**

- implemented `aggregate_on_attribute_value` function
- tests
- documentation

**Depends on** https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33669

### CODE REVIEW

File: processor/transformprocessor/README.md
Comment (odubajDT) [inline processor/transformprocessor/README.md:425]:
```suggestion
```

**Comment by odubajDT** [comment]
Hey @TylerHelmuth @evan-bradley would you please have a look?

File: processor/transformprocessor/README.md
Comment (alecrajeev) [inline processor/transformprocessor/README.md:423]: `aggreates` -> `aggregates`

(4th word in the sentence)

Comment (odubajDT) [inline processor/transformprocessor/README.md:423]:
```suggestion
The `aggregate_on_attribute_value` function aggregates all datapoints in the metric containing the attribute `attribute` with one of the values present in the `values` parameter. `function` is a case-sensitive string that represents the aggregation function.
```

**Comment by odubajDT** [comment]
@TylerHelmuth @evan-bradley would you have time to look into this? Thanks!

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
> Hey @TylerHelmuth @evan-bradley would you please have a look?

Just doubling checking with code owners before it is merged.

File: processor/transformprocessor/README.md
Comment (evan-bradley) [inline processor/transformprocessor/README.md:447]:
```suggestion
- `aggregate_on_attribute_value("sum", "attr1", ["val1", "val2"], "new_val") where name == "system.memory.usage"`
```

File: processor/transformprocessor/README.md
Comment (evan-bradley) [inline processor/transformprocessor/README.md:458]:
```suggestion
   - aggregate_on_attribute_value("sum", "attr1", ["val1", "val2"], "new_val") where name == "system.memory.usage"
```

File: processor/transformprocessor/README.md
Comment (evan-bradley) [inline processor/transformprocessor/README.md:425]:
```suggestion
The `aggregate_on_attribute_value` function aggregates all datapoints in the metric containing the attribute `attribute` with one of the values present in the `values` into a single datapoint where the attribute has the value `newValue`. `function` is a case-sensitive string that represents the aggregation function.
```

I think we could make this a bit more concise by just explaining how the function uses the parameters to operate on the metric.

File: processor/transformprocessor/internal/metrics/func_agregate_on_attribute_value_metrics_test.go
Comment (evan-bradley) [inline processor/transformprocessor/internal/metrics/func_agregate_on_attribute_value_metrics_test.go:19]: Could you also add a test with three datapoints where two are aggregated and one is left unaggregated?

**Comment by MovieStoreGuy** [comment]
Hey @evan-bradley, just double checking you're okay with the changes made?

**Review by evan-bradley — APPROVED** [review]

**Comment by evan-bradley** [comment]
This looks good to me, thanks for continuing to push this along @MovieStoreGuy.

@TylerHelmuth could you take a look when you have time?

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**
